test/run_test.py

```diff
@@ -1372,7 +1372,7 @@ def success_callback(err_message):
         os.environ['PARALLEL_TESTING'] = '1'
         for test in selected_tests_parallel:
             options_clone = copy.deepcopy(options)
-            if test not in PYTEST_BLOCKLIST:
+            if test not in PYTEST_BLOCKLIST and os.getenv('PYTORCH_TEST_DO_NOT_USE_PYTEST', '0') == '0':
```

Comment:
To avoid code duplication, do you mind adding `can_run_in_pytest` predicate that would be use on lines 1375 and 1393
 
```suggestion
            if can_run_in_pytes(test):
```